// @remotion/tailwind/dist/cjs/enable.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.enableTailwind = void 0;
/**
 * @description A function that modifies the default Webpack configuration to make the necessary changes to support Tailwind.
 * @see [Documentation](https://www.remotion.dev/docs/tailwind/enable-tailwind)
 */
exports.enableTailwind = ((currentConfiguration, options) => {
    var _a;
    return {
        ...currentConfiguration,
        module: {
            ...currentConfiguration.module,
            rules: [
                ...(((_a = currentConfiguration.module) === null || _a === void 0 ? void 0 : _a.rules)
                    ? currentConfiguration.module.rules
                    : []).filter((rule) => { var _a; return rule && rule !== '...' && !((_a = rule.test) === null || _a === void 0 ? void 0 : _a.toString().includes('.css')); }),
                {
                    test: /\.css$/i,
                    use: [
                        require.resolve('style-loader'),
                        require.resolve('css-loader'),
                        {
                            loader: require.resolve('postcss-loader'),
                            options: {
                                postcssOptions: {
                                    plugins: [
                                        require.resolve('postcss-preset-env'),
                                        [
                                            require.resolve('tailwindcss'),
                                            (options === null || options === void 0 ? void 0 : options.configLocation)
                                                ? { config: options.configLocation }
                                                : {},
                                        ],
                                        require.resolve('autoprefixer'),
                                    ],
                                },
                            },
                        },
                    ],
                },
            ],
        },
    };
});
