// @remotion/tailwind/dist/esm/index.mjs
import { createRequire } from "node:module";
var __require = /* @__PURE__ */ createRequire(import.meta.url);

// src/enable.ts
var enableTailwind = (currentConfiguration, options) => {
  return {
    ...currentConfiguration,
    module: {
      ...currentConfiguration.module,
      rules: [
        ...((currentConfiguration.module?.rules) ? currentConfiguration.module.rules : []).filter((rule) => rule && rule !== "..." && !rule.test?.toString().includes(".css")),
        {
          test: /\.css$/i,
          use: [
            __require.resolve("style-loader"),
            __require.resolve("css-loader"),
            {
              loader: __require.resolve("postcss-loader"),
              options: {
                postcssOptions: {
                  plugins: [
                    __require.resolve("postcss-preset-env"),
                    [
                      __require.resolve("tailwindcss"),
                      options?.configLocation ? { config: options.configLocation } : {}
                    ],
                    __require.resolve("autoprefixer")
                  ]
                }
              }
            }
          ]
        }
      ]
    }
  };
};
export {
  enableTailwind
};
